cloud_governance/cost_expenditure/generate_cost_explorer_report.py

```diff
@@ -0,0 +1,79 @@
+import json
+from multiprocessing import Process
+
+from cloud_governance.common.aws.cost_explorer.cost_explorer_operations import CostExplorerOperations
+from cloud_governance.common.logger.init_logger import logger
+from cloud_governance.common.elasticsearch.elasticsearch_operations import ElasticSearchOperations
+
+
+class GenerateCostExplorerReport:
+
+    def __init__(self, es_host: str, es_port: str, es_index: str, metric_type: str):
+        self.__cost_explorer = CostExplorerOperations()
+        self.__es_host = es_host
+        self.__es_port = es_port
+        self.__es_index = es_index
+        self.__elastic_search_operations = ElasticSearchOperations(es_host=self.__es_host, es_port=self.__es_port)
+        self.metric_type = metric_type
+
+    def filter_data_by_tag(self, groups: list, tag: str):
+        """
+        This method extract data by tag
+        @param tag:
+        @param groups: Data from the cloud explorer
+        @return: converted into dict format
+        """
+        data = []
+        for group in groups:
+            name = ''
+            amount = ''
+            if group.get('Keys'):
+                name = group.get('Keys')[0].split('$')[-1]
+                name = name if name else 'NoTagKey'
+            if group.get('Metrics'):
+                amount = group.get('Metrics').get(self.metric_type).get('Amount')
+            if name and amount:
+                data.append({tag: name, 'Cost': round(float(amount), 3)})
+        return data
+
+    def __get_daily_cost_by_tags(self):
+        """
+        This method extracts the costs by tags and upload to elastic search
+        @return:
+        """
+        if not self.metric_type:
+            self.metric_type = 'BlendedCost'
```

Comment:
Can you get it from environment variable ?

Reply:
yes, if they didn't provide. By default we will take BlendedCost